shopfloor/services/zone_picking.py

```diff
@@ -765,6 +797,14 @@ def set_destination(
         * set_line_destination+confirmation_required: the scanned location is not
           in the expected one but is valid (in picking type's default destination)
         """
+        if self.work.menu.no_prefill_qty:
+            # While a product, a lot, a packaging …  is scanned, increment the
+            # quantity_done.
+            # Otherwise, fallback to the destination package
+            res = self.scan_source(barcode, quantity, confirmation)
```

Comment:
Why do we call the previous endpoint `scan_source` from `set_destination` here? I do not understand the comment.

Reply:
The step is calling set_destination on scan but you can continue to scan the source to increment the qty of the scanned source instead of scanning a destination

Reply:
After seeing this for way too many times I think is ugly as the other hack. 
The comment does not explain what you typed here.

Reply:
It should be the job of the front end to call the correct endpoint, IMO.